engines/crypto/filesystem/filesystem.go

```diff
@@ -65,123 +68,155 @@ func NewFilesystemPEMEngine(logger *logrus.Entry, conf config.CryptoEngineConfig
+func (engine *FilesystemCryptoEngine) CreateECDSAPrivateKey(curve elliptic.Curve) (string, crypto.Signer, error) {
+	engine.logger.Debugf("creating ECDSA private key")
+
+	_, key, err := software.NewSoftwareCryptoEngine(engine.logger).CreateECDSAPrivateKey(curve)
 	if err != nil {
-		lGo.Errorf("could not create %s ECDSA key: %s", keyID, err)
-		return nil, err
+		engine.logger.Errorf("could not create ECDSA private key: %s", err)
+		return "", nil, err
 	}
 
-	return p.ImportECDSAPrivateKey(key, keyID)
+	engine.logger.Debugf("ECDSA key successfully generated")
+	return engine.importKey(key)
 }
 
-func (p *FilesystemCryptoEngine) DeleteKey(keyID string) error {
-	return os.Remove(p.storageDirectory + "/" + keyID)
+func (engine *FilesystemCryptoEngine) DeleteKey(keyID string) error {
+	return os.Remove(engine.storageDirectory + "/" + keyID)
 }
 
-func (p *FilesystemCryptoEngine) ImportRSAPrivateKey(key *rsa.PrivateKey, keyID string) (crypto.Signer, error) {
-	lGo.Debugf("importing RSA %d key for keyID: %s", key.Size(), keyID)
-	p.checkAndCreateStorageDir()
+func (engine *FilesystemCryptoEngine) ImportRSAPrivateKey(key *rsa.PrivateKey) (string, crypto.Signer, error) {
+	engine.logger.Debugf("importing RSA private key")
 
-	err := os.WriteFile(p.storageDirectory+"/"+keyID, pem.EncodeToMemory(&pem.Block{
-		Bytes: x509.MarshalPKCS1PrivateKey(key),
-		Type:  "RSA PRIVATE KEY",
-	}), 0600)
+	keyID, signer, err := engine.importKey(key)
 	if err != nil {
-		lGo.Errorf("could not save %s RSA key: %s", keyID, err)
-		return nil, err
+		engine.logger.Errorf("could not import RSA key: %s", err)
+		return "", nil, err
 	}
 
-	return p.GetPrivateKeyByID(keyID)
+	engine.logger.Debugf("RSA key successfully imported")
+	return keyID, signer, nil
 }
 
-func (p *FilesystemCryptoEngine) ImportECDSAPrivateKey(key *ecdsa.PrivateKey, keyID string) (crypto.Signer, error) {
-	lGo.Debugf("importing ECDSA %d key for keyID: %s", key.Params().BitSize, keyID)
-	p.checkAndCreateStorageDir()
+func (engine *FilesystemCryptoEngine) ImportECDSAPrivateKey(key *ecdsa.PrivateKey) (string, crypto.Signer, error) {
+	engine.logger.Debugf("importing ECDSA private key")
 
-	b, err := x509.MarshalECPrivateKey(key)
+	keyID, signer, err := engine.importKey(key)
 	if err != nil {
-		return nil, err
+		engine.logger.Errorf("could not import ECDSA key: %s", err)
+		return "", nil, err
 	}
 
-	err = os.WriteFile(p.storageDirectory+"/"+keyID, pem.EncodeToMemory(&pem.Block{
-		Bytes: b,
-		Type:  "EC PRIVATE KEY",
-	}), 0600)
+	engine.logger.Debugf("ECDSA key successfully imported")
+	return keyID, signer, nil
+}
+
+func (engine *FilesystemCryptoEngine) importKey(key interface{}) (string, crypto.Signer, error) {
+	var pubKey any
+	switch k := key.(type) {
+	case *rsa.PrivateKey:
+		pubKey = &k.PublicKey
+	case *ecdsa.PrivateKey:
+		pubKey = &k.PublicKey
+	default:
+		return "", nil, errors.New("unsupported key type")
+	}
+
+	softEngine := software.NewSoftwareCryptoEngine(engine.logger)
```

Comment:
Same than previous comments. Can we add this as a property at the FileSystemCryptoEngine?